### PR TITLE
Scaffold a basic argument parsing structure with clap.

### PR DESCRIPTION
**Pull request by reillysiemens**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,3 +1,73 @@
...
+
 fn main() {
-    println!("Hello, world!");
+    let app = clap_app!(myapp =>
+        (version: "0.1.0")
+        (author: "Bike Barn <https://github.com/bike-barn/hermit>")
+        (about: "A home directory configuration management assistant.")
+        (@subcommand add =>
+            (about: "Add files to your hermit shell")
+            (usage: "hermit add [[-r|--redact=] <filepattern>…]")
```
Comment (RadicalZephyr) [inline src/main.rs]: shouldn't be a redact option.  that's not even on the issue list yet

Comment (reillysiemens) [inline src/main.rs]: Ah. Right. Options aren't being enforced through subcommand arguments yet, but I'll update the usage to reflect that.

File: src/main.rs [hunk 2]
```diff
@@ -1,3 +1,73 @@
...
+            (about: "Permanently remove a hermit shell")
+            (usage: "hermit nuke <shell-name>")
+        )
+        (@subcommand status =>
+            (about: "Display the status of your hermit shell")
+            (usage: "hermit status")
+        )
+        (@subcommand use =>
+            (about: "Switch to using a different hermit shell")
+            (usage: "hermit use [shell-name]")
```
Comment (RadicalZephyr) [inline src/main.rs]: Shouldn't the shell-name be required here? or did we have a "default" shell name fallback?

Comment (reillysiemens) [inline src/main.rs]: We did have a `default` shell name fallback previously.

**Comment by RadicalZephyr** [comment]
Cool.  Once the redact option reference is removed this is good to go.

**Pull request merged**